Synergy/Controllers/RandomCodeController.cs

```diff
@@ -0,0 +1,76 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Synergy.Authorization;
+using Synergy.Models;
+using Synergy.Services;
+using Permissions = Synergy.Authorization.Permissions;
+
+namespace Synergy.Controllers;
+
+[ApiController]
+[Route("/api/[controller]")]
+public class RandomCodeController : ControllerBase
+{
+    private readonly RandomCodeService _randomCodeService;
+
+    public RandomCodeController(RandomCodeService randomCodeService)
+    {
+        _randomCodeService = randomCodeService;
+    }
+
+    [HttpPost("generate-code")]
+    [Authorize]
+    [AuthorizePermission(Permissions.CanRegisterTeam)]
+    public async Task<IActionResult> GenerateCode([FromBody] TeamId body)
```

Comment:
You don't have any validation on this "body" variable. What if it's empty or not provided?